components/sup/src/manager/file_watcher.rs

```diff
@@ -1141,56 +1141,33 @@ pub struct FileWatcher<C: Callbacks, W: Watcher> {
 
 /// Convenience function for returning a new file watcher that matches
 /// the platform.
-pub fn default_file_watcher<P, C>(path: P,
-                                  callbacks: C)
-                                  -> Result<FileWatcher<C, RecommendedWatcher>>
+pub fn create_file_watcher<P, C>(path: P,
+                                 callbacks: C,
+                                 ignore_initial: Option<bool>)
+                                 -> Result<FileWatcher<C, SupWatcher>>
     where P: Into<PathBuf>,
           C: Callbacks
 {
-    FileWatcher::<C, RecommendedWatcher>::create(path, callbacks)
-}
-
-pub fn default_file_watcher_with_no_initial_event<P, C>(
-    path: P,
-    callbacks: C)
-    -> Result<FileWatcher<C, RecommendedWatcher>>
-    where P: Into<PathBuf>,
-          C: Callbacks
-{
-    FileWatcher::<C, RecommendedWatcher>::create_with_no_initial_event(path, callbacks)
+    FileWatcher::<C, SupWatcher>::create(path, callbacks, ignore_initial)
 }
 
 impl<C: Callbacks, W: Watcher> FileWatcher<C, W> {
     /// Creates a new `FileWatcher`.
     ///
     /// This will create an instance of `W` and start watching the
     /// paths. When looping the file watcher, it will emit an initial
-    /// "file appeared" event if the watched file existed when the
-    /// file watcher was created.
-    ///
-    /// Will return `Error::NotifyCreateError` if creating the watcher
-    /// fails. In case of watching errors, it returns
-    /// `Error::NotifyError`.
-    pub fn create<P>(path: P, callbacks: C) -> Result<Self>
-        where P: Into<PathBuf>
-    {
-        Self::create_instance(path, callbacks, true)
-    }
-
-    /// Creates a new `FileWatcher`.
-    ///
-    /// This will create an instance of `W` and start watching the
-    /// paths. When looping the file watcher, it will not emit any
-    /// initial "file appeared" event even if the watched file existed
-    /// when the file watcher was created.
-    ///
+    /// "file appeared" event if ignore_initial is Some(false) and the
+    /// watched file existed when the file watcher was created.
     /// Will return `Error::NotifyCreateError` if creating the watcher
     /// fails. In case of watching errors, it returns
     /// `Error::NotifyError`.
-    pub fn create_with_no_initial_event<P>(path: P, callbacks: C) -> Result<Self>
+    pub fn create<P>(path: P, callbacks: C, ignore_initial: Option<bool>) -> Result<Self>
```

Comment:
Why is this `Option<bool>` when a `bool` would suffice. There are only 2 possible code paths. Option<bool> implies 3 code paths, `Some(true)`, `Some(false)` and `None`.

Reply:
I originally was thinking of optional parameters in other languages and associated default behavior.  That does not apply here - will refactor to use bool.